plugins/orchestrator/src/components/ExecuteWorkflowPage/ExecuteWorkflowPage.tsx

```diff
@@ -27,154 +30,187 @@ import {
+        </Box>
+      </Grid>
+      <Grid item xs={12}>
+        <SubmitBtn
+          submitting={isExecuting}
+          handleClick={() => handleExecute(getParameters)}
+        >
+          Run
+        </SubmitBtn>
+      </Grid>
+    </Grid>
+  );
+};
+
+export const ExecuteWorkflowPage = () => {
   const orchestratorApi = useApi(orchestratorApiRef);
   const { workflowId } = useRouteRefParams(executeWorkflowRouteRef);
   const { businessKey } = useRouteRefParams(
     executeWorkflowWithBusinessKeyRouteRef,
   );
-  const [loading, setLoading] = useState(false);
-  const [schemaResponse, setSchemaResponse] =
-    useState<WorkflowDataInputSchemaResponse>();
-  const [workflowDialogOpen, setWorkflowDialogOpen] = useState<boolean>(false);
-  const [formState, setFormState] = useState(props.initialState);
-
+  const [isExecuting, setIsExecuting] = useState(false);
+  const [updateError, setUpdateError] = React.useState<Error>();
   const navigate = useNavigate();
   const instanceLink = useRouteRef(workflowInstanceRouteRef);
 
-  useEffect(() => {
-    setLoading(true);
-    orchestratorApi
-      .getWorkflowDataInputSchema(workflowId)
-      .then(response => {
-        setSchemaResponse(response);
-      })
-      .finally(() => {
-        setLoading(false);
-      });
-  }, [orchestratorApi, workflowId]);
+  const {
+    value: schemaResponse,
+    loading,
+    error: responseError,
+  } = useAsync(
+    async (): Promise<WorkflowDataInputSchemaResponse> =>
+      await orchestratorApi.getWorkflowDataInputSchema(workflowId),
+    [orchestratorApi, workflowId],
+  );
 
-  const onExecute = useCallback(async () => {
-    const parameters: Record<string, JsonValue> = {};
-    if (schemaResponse?.schema && formState) {
-      for (const key in formState) {
-        if (formState.hasOwnProperty(key)) {
-          const property = formState[key];
-          Object.assign(parameters, { [key]: property });
-        }
+  const handleExecute = useCallback(
+    async (getParameters: () => Record<string, JsonValue>) => {
+      setUpdateError(undefined);
+      let parameters: Record<string, JsonValue> = {};
+      try {
+        parameters = getParameters();
+      } catch (err) {
+        setUpdateError(getErrorObject(err));
+        return;
       }
-    }
-
-    setLoading(true);
-    const workflowCategory = getWorkflowCategory(
-      schemaResponse?.workflowItem.definition,
-    );
-    if (workflowCategory === WorkflowCategory.ASSESSMENT) {
-      Object.assign(parameters, { businessKey: crypto.randomUUID() });
-    } else {
-      Object.assign(parameters, { businessKey: businessKey ?? '' });
-    }
-    const response = await orchestratorApi.executeWorkflow({
+      try {
+        const workflowCategory = getWorkflowCategory(
+          schemaResponse?.workflowItem.definition,
+        );
+        if (workflowCategory === WorkflowCategory.ASSESSMENT) {
+          Object.assign(parameters, { businessKey: crypto.randomUUID() });
```

Comment:
We need to remove this part, the BK will be generated in the BE.
See https://issues.redhat.com/browse/FLPATH-842

Reply:
@anludke, @richardW98